theano/gof/fg.py

```diff
@@ -664,17 +665,24 @@ def orderings(self):
                                     str(feature.orderings) +
                                     ". Nondeterministic object is " +
                                     str(orderings))
+                if len(orderings) > 0:
+                    all_orderings.append(orderings)
+                    for node, prereqs in iteritems(orderings):
+                        if not isinstance(prereqs, (list, OrderedSet)):
+                            raise TypeError(
+                                "prereqs must be a type with a "
+                                "deterministic iteration order, or toposort "
+                                " will be non-deterministic.")
+        if len(all_orderings) == 1:
+            # If there is only 1 ordering, we reuse it directly.
+            return all_orderings[0]
```

Comment:
I checked. DestroyHandler return directly the internal structure that will be reused. So I think we can use the normal copy:

return all_orderings[0].copy()